src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -564,13 +664,78 @@ class CreateMaterializedViewStatementSegment(BaseSegment):
     )
 
 
+class CreateDatabaseStatementSegment(ansi.CreateDatabaseStatementSegment):
+    """A `CREATE DATABASE` statement.
+
+    As specified in
+    https://clickhouse.com/docs/en/sql-reference/statements/create/database
+    """
+
+    type = "create_database_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        "DATABASE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("DatabaseReferenceSegment"),
+        AnySetOf(
+            Sequence(
+                "ON",
+                "CLUSTER",
+                Ref("ExpressionSegment"),
+                optional=True,
+            ),
+            Ref("DatabaseEngineSegment", optional=True),
+            Sequence(
+                "COMMENT",
+                Ref("ExpressionSegment"),
+                optional=True,
+            ),
+            Sequence(
+                "SETTINGS",
+                Delimited(
+                    AnyNumberOf(
+                        Sequence(
+                            Ref("NakedIdentifierSegment"),
+                            Ref("EqualsSegment"),
+                            OneOf(
+                                Ref("NakedIdentifierSegment"),
+                                Ref("NumericLiteralSegment"),
+                                Ref("QuotedLiteralSegment"),
+                                Ref("BooleanLiteralGrammar"),
+                            ),
+                            optional=True,
+                        ),
+                    )
+                ),
+                optional=True,
+            ),
+        ),
+        AnyNumberOf(
+            "TABLE",
+            "OVERRIDE",
+            Ref("TableReferenceSegment"),
+            Bracketed(
+                Delimited(
+                    OneOf(
```

Comment:
Delimited has this OneOf functionality, so here it isn't needed

Reply:
Fixed in commit `f56765e`.